### ISSUE TITLE
github action workflow to add CI

### ISSUE THREAD

**Issue by jatin69** [shoaibrayeen/Coding-Practice#2]
I see that you're trying to add CI for this project. 
I'd like to highlight a few things.

- workflows are supposed to test (primarily), but in projects without tests. we can simply use them to test if everything compiles.
- to compile these many files, you'll need to write a decent makefile
- and this assumes all files are c++ if some random folders contain some random files, it's gonna be hard to judge programmatically which files to compile with g++ and which file with python

So there are several things to be done before the workflow can be made active.

- segregate different languages if any into separate folders, ideally top-level folders. Language folders inside topic folders won't work.
- then write a makefile to compile all those files and languages. A good makefile will compile everything in a directory ( `*.cpp` and `*.py`) respectively
- then edit the workflow file to simply make everything, remove the configure step because there's nothing to configure in our case. simply run make command.

Let me know if you need any help here.

**Comment by jatin69** [comment]
@shoaibrayeen okay, I spent some time. There are several other problems as well.

- make tool does not support files and folders with spaces and breaks down when encounters any such file or folder
- to correct that I renamed all files with bash shell to remove spaces
```shell
find -iname .\*\* | rename -v 's/ /_/g' *\ *
find . -type f -name "* *.*" -exec rename "s/\s/_/g" {} \;
find . -type f -name "* *" -exec rename "s/\s/_/g" {} \;
find . -name "* *" -exec rename "s/\s/_/g" {} \;
```
- turns out that's not all. some file name even has special characters. `()` and `[]` for example. Further removing them brings into light a bigger problem

- A lot of code files are not compilable. Small unnoticed bugs. Non-standard code formulations. Basically sloppiness in a lot of places. 

I have removed sloppiness from a few files but the quantity is too large. You'll have to invest time in correcting them all by looking at the errors.
